src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -52,21 +52,27 @@ public static void SetAccessControl(this FileInfo fileInfo, FileSecurity fileSec
 
         public static FileSecurity GetAccessControl(this FileStream fileStream)
         {
+            if (fileStream == null)
+                throw new NullReferenceException(nameof(fileStream));
```

Comment:
After looking at other extension methods, I noticed we don't use `NullReferenceException` for `this` parameters. We use `ArgumentNullException` instead. Examples:

- [corefx/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchive.Create.cs](https://github.com/dotnet/corefx/blob/master/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchive.Create.cs)

- [corefx/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs](https://github.com/dotnet/corefx/blob/master/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs)

Also, do you mind adding the exception to the triple slash comments?

Reply:
Gotcha. This was a source of confusion.  The tests will also need to be updated as they are expecting NullReferenceException.  I'll put all the changes into a PR for the new repo using the ZipArchive extensions as reference.

Thanks!

Reply:
Sure, if you want to submit the PR in the new repo, that's fine too. Thank you again.